syntax = "proto3";
option cc_enable_arenas = true;

import "ydb/public/api/protos/annotations/validation.proto";
import "ydb/public/api/protos/ydb_issue_message.proto";
import "ydb/public/api/protos/ydb_operation.proto";
import "ydb/public/api/protos/ydb_scheme.proto";

package Ydb.Replication;
option java_package = "com.yandex.ydb.replication";

message DescribeReplicationRequest {
    Ydb.Operations.OperationParams operation_params = 1;
    // Replication path.
    string path = 2 [(required) = true];
}

message DescribeReplicationResponse {
    // Result of request will be inside operation.
    Ydb.Operations.Operation operation = 1;
}

message ConnectionParams {
    message StaticCredentials {
        string user = 1;
        string password_secret_name = 2;
    }

    message OAuth {
        string token_secret_name = 1;
    }

    string endpoint = 1;
    string database = 2;

    oneof credentials {
        StaticCredentials static_credentials = 3;
        OAuth oauth = 4;
    }
}

message DescribeReplicationResult {
    message Item {
        string source_path = 1;
        string destination_path = 2;
        optional string source_changefeed_name = 3;
        uint64 id = 4;
    }

    message RunningState {
    }

    message ErrorState {
        repeated Ydb.Issue.IssueMessage issues = 1;
    }

    message DoneState {
    }

    // Description of scheme object.
    Ydb.Scheme.Entry self = 1;
    ConnectionParams connection_params = 2;
    repeated Item items = 3;
    oneof state {
        RunningState running = 4;
        ErrorState error = 5;
        DoneState done = 6;
    }
}

